### PR TITLE
roxdebug: Start dlv debugger from writable directory

### PR DESCRIPTION
**Pull request by mclasmeier**
Otherwise it will fail, because it seems to require a writable current-working-directory.

### CODE REVIEW

**Review by vikin91 — APPROVED** [review]
Have not tested that, but I remember struggling with remote debugging recently. Does it work for you after that change?

**Comment by mclasmeier** [comment]
@vikin91 This seems to be working for attaching to dlv attached to sensor.

File: scripts/dev/roxdebug.sh [hunk 1]
```diff
@@ -53,6 +53,73 @@ fi
...
+    read_only_root_fs=$(kubectl -n stackrox get "${deployment}" \
+      -o=jsonpath="{.spec.template.spec.containers[?(@.name=='${container_name}')].securityContext.readOnlyRootFilesystem}")
+    if [[ "${read_only_root_fs}" == "true" ]]; then
+      echo "Configuring root filesystem of container $container_name within ${deployment} to be read-write."
+      kubectl -n stackrox patch "${deployment}" \
+        -p '{"spec":{"template":{"spec":{"containers":[{"name":"sensor","securityContext":{"readOnlyRootFilesystem":false}}]}}}}'
+      # Unfortunately kubectl wait cannot wait yet for creation of new resources (pods, in this case).
+      # Hence, as a workaround to prevent a race here we need to wait a couple of seconds.
+      sleep 5
+      kubectl -n stackrox wait --for=condition=Ready pod -l app=sensor --timeout=60s
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply scripts/dev/roxdebug.sh:118]: Thanks for catching it.

File: scripts/dev/roxdebug.sh [hunk 2]
```diff
@@ -82,7 +80,7 @@ case "${deployment_name}" in
     container_name="indexer"
     ;;
   *)
-    echo >2 "Unknown deployment '$deployment', cannot tell which container is to be debugged."
+    echo >&2 "Unknown deployment '$deployment', cannot tell which container is to be debugged."
```
Comment (vikin91) [inline scripts/dev/roxdebug.sh:83]: Good catch! I missed that!

File: scripts/dev/roxdebug.sh [hunk 3]
```diff
@@ -99,23 +97,26 @@ echo
...
+    read_only_root_fs="$(kubectl -n stackrox get "${deployment}" \
+      -o=jsonpath="{.spec.containers[?(@.name=='${container_name}')].securityContext.readOnlyRootFilesystem}")"
     if [[ $read_only_root_fs == "true" ]]; then
       echo >&2 "Cannot debug ${deployment}: the pod uses a read-only root filesystem"
       exit 1
     fi
   else
-    read_only_root_fs=$(kubectl -n stackrox get "${deployment}" \
-      -o=jsonpath="{.spec.template.spec.containers[?(@.name=='${container_name}')].securityContext.readOnlyRootFilesystem}")
+    read_only_root_fs=$(kubectl -n stackrox get "${deployment}" -o=jsonpath="{.spec.template.spec.containers[?(@.name=='${container_name}')].securityContext.readOnlyRootFilesystem}")
```
Comment (vikin91) [inline scripts/dev/roxdebug.sh:107]: This one is still unquoted, but that is optional. I think the chance of receiving here a string containing spaces is low.

**Review by vikin91 — APPROVED** [review]
The changes look good! I didn't have occasion to test it myself, so I trust that you checked it.
My comments are non-blocking and it is fine to resolve them

File: scripts/dev/roxdebug.sh (hunk 3 above)
Comment (mclasmeier) [reply scripts/dev/roxdebug.sh:107]: I can do that fort consistency, but, out of curiosity, is it actually required -- I mean, even assuming spaces to appear here. Shellcheck did not complain, and, according to my tests:

```
$ bash
$ foo=$(echo foo bar)
$ echo "'$foo'"
'foo bar'
$
```

Comment (vikin91) [reply scripts/dev/roxdebug.sh]: Indeed it is my rule of a thumb that is not always required. I am unable to give you a quick example currently that would show the dangers of not quoting. But I can share one when I find/recall one :)

**Pull request merged**